pkg/kubectl/resource/selector.go

```diff
@@ -76,12 +84,72 @@ func (r *Selector) Visit(fn VisitorFunc) error {
 
 		Object:          list,
 		ResourceVersion: resourceVersion,
+		Dynamic:         r.Dynamic,
 	}
 	return fn(info, nil)
 }
 
+func (r *Selector) dynamicClient() (*dynamic.Client, error) {
+	rc, ok := r.Client.(*restclient.RESTClient)
+	if !ok {
+		return nil, errors.New("could not build dynamic client from RESTClient")
+	}
+
+	dc, err := dynamic.FromRESTClient(rc)
+	if err != nil {
+		return nil, err
+	}
+
+	return dc.ParameterCodec(dynamic.VersionedParameterEncoderWithV1Fallback), nil
+}
+
+func (r *Selector) list() (runtime.Object, error) {
+	if !r.Dynamic {
```

Comment:
This seems very wrong - Helper is *already* supposed to be generic - doing this is redundant.